packages/aws-cdk/lib/commands/init/init.ts

```diff
@@ -84,16 +91,27 @@ export async function cliInit(options: CliInitOptions) {
   const generateOnly = options.generateOnly ?? false;
   const workDir = options.workDir ?? process.cwd();
 
+  // Validate that if template-path is provided, exactly one custom template source must be provided
+  if (options.templatePath && !options.fromPath && !options.fromGitUrl) {
+    throw new ToolkitError('--template-path can only be used with --from-path or --from-git-url');
```

Comment:
why not use a check in yargs instead? In general, try to catch problems as early as you can.